PHPCS fixes for 1.9.1-dev

paulschreiber opened the pull request.

paulschreiber commented:
Two lines generate escaping warnings:
* class-wpcom-liveblog-lazyloader.php:142
* class-wpcom-liveblog-amp-template.php:97

Please inspect them and choose an appropriate escaping function, or add a `phpcs:ignore` comment.

GaryJones commented on composer.json:
```diff
@@ -5,6 +5,9 @@
     "type"       : "wordpress-plugin",
     "license"    : "GPL-2.0+",
     "require"    : {
+        "wp-coding-standards/wpcs": "^1.0.0",
```
These should be `require-dev` dependencies.

paulschreiber replied:
fixed with 268daa4.

GaryJones commented on phpcs.xml:
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0"?>
```
This should be named `.phpcs.xml.dist` or `phpcs.xml.dist`.

paulschreiber replied:
I based this on the work done at https://github.com/Automattic/amp-wp/ cc @westonruter

GaryJones replied:
@westonruter should know better :-)

I've created https://github.com/Automattic/amp-wp/issues/1324 for them to update it at AMP.

paulschreiber replied:
Renamed with 5767fb0.

GaryJones commented on phpcs.xml:
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0"?>
+<ruleset name="WordPress Coding Standards for Liveblog">
```
Name should be `"Custom ruleset for Liveblog"`. It's a ruleset, not a standard.

paulschreiber replied:
Also copied from amp-wp.

GaryJones replied:
This is just about being accurate and educating developers.

For your reference (since I see you on many different repos doing PRs improving PHPCS setups like this - and thank you for that!), here's my [plugin boilerplate `.phpcs.xml.dist`](https://github.com/GaryJones/plugin-boilerplate/blob/master/.phpcs.xml.dist) file, which applies many good practices.

paulschreiber replied:
Fixed with 14a4a9e.

GaryJones commented on phpcs.xml:
```diff
@@ -0,0 +1,14 @@
+<ruleset name="WordPress Coding Standards for Liveblog">
+
+	<rule ref="WordPress-VIP" />
+	<rule ref="WordPress-Extra" />
```
There should be no spaces before closing `/>`. This is XML, not XHTML.

paulschreiber replied:
Also copied from amp-wp.

paulschreiber replied:
Fixed with 0ca4449.

GaryJones commented on phpcs.xml:
```diff
@@ -0,0 +1,14 @@
+	<rule ref="WordPress-VIP" />
+	<rule ref="WordPress-Extra" />
+
+	<arg value="s"/>
```
Include the progress (`p`), colors, extensions, basepath and parallel args - this plugin is PHP 5.6 and above, so PHPCS 3.x will be installed (3.3.1 currently).

paulschreiber replied:
I was previously told not to use `--colors`:
https://github.com/WordPress/gutenberg/pull/1685#issuecomment-312780011

paulschreiber replied:
`extensions` is already present.

paulschreiber replied:
What do you want for `basepath` and `parallel`? @westonruter any thoughts here?

paulschreiber replied:
Fixed based on your boilerplate. 15df6b2.

GaryJones replied:
`basepath="."` is enough to remove everything before the current directory.
`parallel="8"` is a good default, though I put it to 50 on my plugins - I've not yet encountered a drawback, though that doesn't mean there isn't one.

GaryJones commented on phpcs.xml.dist:
```diff
@@ -1,8 +1,9 @@
 <ruleset name="Custom ruleset for Liveblog">
 
-	<rule ref="WordPress-VIP"/>
+	<rule ref="WordPress-Core"/>
```
No need to include `WordPress-Core` if including `WordPress-Extra`, as Extra already includes Core.

paulschreiber commented:
Updated to use `WordPress` as the ruleset.

paulschreiber commented:
Failure is a Travis error:
```
The command "bash bin/install-wp-tests.sh wordpress_test root '' localhost $WP_VERSION" failed and exited with 7 during .
```

paulschreiber commented:
First, you complained about the spacing in the XML tags. You did not mention anything about the ruleset. Then, I updated the rulesets to Core+Extra+VIPMinimum. You said to remove Core. Then, I changed the rulesets to WordPress+VIPMinimum, and you said not to use WordPress.

This was a ridiculous amount of back-and-forth for two lines of XML.

It would have been better if you had said up front:
> Use the `WordPress-Extra` and `WordPressVIPMinimum` rulesets. Do not use `WordPress` or `WordPress-Core`. You may use `WordPress-Docs` if you want.

or…accepted the pull request and made the change yourself.

paulschreiber commented:
@philipjohn I have updated this for the released version of 1.9. I caught many more errors. Please run `phpcs` on the code before releasing.

GaryJones commented:
@paulschreiber 	Thanks for the new commits.

Within the File Changes, I see a whole load of bits that aren't related to CS - changes in version numbers and so on.

Could you please try rebasing on to `master` again?

If that works it should greatly reduce the amount of code to review, and the number of commits.

Thanks!

GaryJones commented on liveblog.php:
```diff
@@ -838,7 +838,7 @@ public static function get_entries_paged( $page, $last_known_entry = false, $id
 			if ( false === $page && false !== $id ) {
 				$index = array_search( (int) $id, array_keys( $entries ), true );
-				$index = $index++;
+				$index++;
```
While the change probably makes sense, I don't think this is something that PHPCS picks up as a violation, is it? (Happy to be shown I'm wrong!)

paulschreiber replied:
PHPCS flagged `$index = $index + 1`. I half-fixed it. Was finishing the fix.

paulschreiber commented:
I merged master back in. Viewing the changes without whitespace-only diffs looks good:
https://github.com/Automattic/liveblog/pull/486/files?w=1

GaryJones commented on classes/class-wpcom-liveblog-wp-cli.php:
```diff
@@ -59,15 +59,17 @@ public function fix_archive( $args, $assoc_args ) {
 
 			// find correct comment_ids to replace incorrect meta_values
-			$correct_ids_array = $wpdb->get_results(
+			$correct_ids_array = $wpdb->get_results( // phpcs:ignore
```
Can you list all of the checks being ignored, rather than a blanket ignore?

GaryJones commented on classes/class-wpcom-liveblog-wp-cli.php:
```diff
@@ -101,9 +103,10 @@ public function fix_archive( $args, $assoc_args ) {
+									$wpdb->update( // phpcs:ignore
 										$wpdb->commentmeta,
 										array( 'meta_value' => $correct_ids[ $i ] ),
+                                        // phpcs:ignore WordPress.DB.SlowDBQuery.slow_db_query_meta_value
```
This line has space indentation, not tabs.

GaryJones commented on classes/class-wpcom-liveblog-wp-cli.php:
```diff
@@ -101,9 +103,10 @@ public function fix_archive( $args, $assoc_args ) {
 								// If this isnt a dry run we can run the database Update.
 								if ( false === $is_dryrun ) {
-									$wpdb->update(
+									$wpdb->update( // phpcs:ignore
```
List all of the checks being ignored.

GaryJones commented on classes/class-wpcom-liveblog-wp-cli.php:
```diff
@@ -59,15 +59,17 @@ public function fix_archive( $args, $assoc_args ) {
 				$wpdb->prepare(
 					"SELECT comment_id FROM $wpdb->comments
 				 WHERE comment_post_id = %d AND comment_id NOT IN
 				 ( SELECT $wpdb->commentmeta.comment_id FROM $wpdb->commentmeta
```
This line, and the next four lines, have some mixed tabs + space indentations.

GaryJones commented on classes/class-wpcom-liveblog-wp-cli.php:
```diff
@@ -114,26 +117,28 @@ public function fix_archive( $args, $assoc_args ) {
 
 			// find comment_ids object with correct content for replacement
-			$correct_contents = $wpdb->get_results(
+			$correct_contents = $wpdb->get_results( // phpcs:ignore
```
List all of the checks being ignored.

GaryJones commented on classes/class-wpcom-liveblog-wp-cli.php:
```diff
@@ -114,26 +117,28 @@ public function fix_archive( $args, $assoc_args ) {
 
 			// find comment_ids that NEED to be replaced
-			$entries_replace = $wpdb->get_results(
+			$entries_replace = $wpdb->get_results( // phpcs:ignore
```
List all of the checks being ignored.

GaryJones commented on classes/class-wpcom-liveblog-wp-cli.php:
```diff
@@ -151,7 +156,7 @@ public function fix_archive( $args, $assoc_args ) {
 
 					if ( false === $is_dryrun ) {
-						$wpdb->update(
+						$wpdb->update( // phpcs:ignore
```
List all of the checks being ignored.

GaryJones commented on composer.json:
```diff
@@ -4,7 +4,10 @@
     "homepage"   : "https://github.com/Automattic/liveblog",
```
In terms of code standards, this whole file should be tab-indented.

GaryJones commented on liveblog.php:
```diff
@@ -1625,11 +1630,11 @@ private static function status_header_with_message( $status, $message ) {
 			$status                       = absint( $status );
 			$official_message             = isset( $wp_header_to_desc[ $status ] ) ? $wp_header_to_desc[ $status ] : '';
-			$wp_header_to_desc[ $status ] = self::sanitize_http_header( $message );
+			$wp_header_to_desc[ $status ] = self::sanitize_http_header( $message ); // phpcs:ignore WordPress.WP.GlobalVariablesOverride.OverrideProhibited
```
I've never even seen or heard of this global before! TIL.

GaryJones commented on phpcs.xml.dist:
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0"?>
+<ruleset name="Custom ruleset for Liveblog">
+
+	<rule ref="WordPress-Extra"/>
```
Also add `WordPress-Docs`? It's not fully complete, but it's a good start for what it does cover. Could leave for a separate PR.

philipjohn replied:
Yep, let's do a separate issue outside  of this PR :+1:

GaryJones commented on phpcs.xml.dist:
```diff
@@ -0,0 +1,17 @@
+	<arg name="parallel" value="50"/>
+	<arg name="extensions" value="php"/>
+	<file>.</file>
+
```
Would like to see text domain checks enabled, and prefixes, and some link references.

See https://github.com/GaryJones/plugin-boilerplate/blob/master/.phpcs.xml.dist, and consider copying this, and amending it with the liveblog bits (no need to include Neutron).

PHPCompatibilityWP standard would also then need to be added to `composer.json` and Travis config.

philipjohn replied:
Let's leave this for a separate change

paulschreiber commented:
@philipjohn I've addressed the issues @GaryJones flagged.
